### PR TITLE
support code in cctor

### PR DESCRIPTION
**Pull request by lightszero**
before only can init const value in cctor step.
``` c#
     static byte[] callscript = HexToBytes("112233aa....");
```

now we can run code like:
``` c#
  static byte[] callscript = ExecutionEngine.EntryScriptHash;
```

add unittest:Test_StaticVarInit

### CODE REVIEW

File: tests/Neo.Compiler.MSIL.UnitTests/UnitTest_StaticVar.cs [hunk 1]
```diff
@@ -20,5 +20,28 @@ public void Test_StaticVar()
...
+        [TestMethod]
+        public void Test_StaticVarInit()
+        {
+            ByteArray var1;
+            ByteArray var2;
+            {
+                var testengine = new TestEngine();
+                testengine.AddEntryScript("./TestClasses/Contract_StaticVarInit.cs");
+                var result = testengine.ExecuteTestCaseStandard("staticinit");
+                //test (1+5)*7 == 42
```
Comment (shargon) [inline tests/Neo.Compiler.MSIL.UnitTests/UnitTest_StaticVar.cs]: What this means?

File: src/Neo.Compiler.MSIL/MSIL/CctorSubVM.cs [hunk 2]
```diff
@@ -100,7 +101,9 @@ public static void Parse(ILMethod from, NeoModule to)
                             }
                             else
                             {
-                                throw new Exception("only byte[] can be defined in here.");
+                                constValue = false;
+                                continue;
+                                //throw new Exception("only byte[] can be defined in here.");
```
Comment (shargon) [inline src/Neo.Compiler.MSIL/MSIL/CctorSubVM.cs]: If now it doesn't throw, we should remove the comment

File: tests/Neo.Compiler.MSIL.UnitTests/UnitTest_StaticVar.cs (hunk 1 above)
Comment (lightszero) [reply tests/Neo.Compiler.MSIL.UnitTests/UnitTest_StaticVar.cs]: call staticinit and directget and check returnvalues equal
```c#

    static byte[] callscript = ExecutionEngine.EntryScriptHash;

        public static object Main(string method, object[] args)
        {
            if (method == "staticinit")
                return testStaticInit();
            if (method == "directget")
                return ExecutionEngine.EntryScriptHash;
            return null;
        }

```

File: src/Neo.Compiler.MSIL/MSIL/CctorSubVM.cs (hunk 2 above)
Comment (lightszero) [reply src/Neo.Compiler.MSIL/MSIL/CctorSubVM.cs]: yes

File: tests/Neo.Compiler.MSIL.UnitTests/UnitTest_StaticVar.cs (hunk 1 above)
Comment (lightszero) [reply tests/Neo.Compiler.MSIL.UnitTests/UnitTest_StaticVar.cs]: error comment,fix it.

File: tests/Neo.Compiler.MSIL.UnitTests/Utils/BuildScript.cs [hunk 3]
```diff
@@ -59,18 +59,23 @@ public void Build(Stream fs, Stream fspdb)
 
             converterIL = new ModuleConverter(log);
             ConvOption option = new ConvOption();
+#if NDEBUG
```
Comment (shargon) [inline tests/Neo.Compiler.MSIL.UnitTests/Utils/BuildScript.cs]: Does we need this directives?

Comment (lightszero) [reply tests/Neo.Compiler.MSIL.UnitTests/Utils/BuildScript.cs]: yes it could be got more clear information when vm error

**Pull request merged**